src/charm.py

```diff
@@ -338,7 +330,7 @@ def _render_conf_file(self) -> str:
             consent_url=self._get_login_ui_endpoint_info("consent_url"),
             error_url=self._get_login_ui_endpoint_info("oidc_error_url"),
             login_url=self._get_login_ui_endpoint_info("login_url"),
-            hydra_public_url=self._public_url,
+            hydra_public_url=self._public_url or f"http://127.0.0.1:{HYDRA_PUBLIC_PORT}/",
```

Comment:
Let me know if I should put the `http://127.0.0.1:4444` into the config file `hydra.yaml`.

Reply:
I think it's better to have it here than in the config file. Is this a necessary change though?

Reply:
The `self: issuer` and [public](https://github.com/canonical/hydra-operator/blob/5292f086517bf7da6206e6212d661330d65225b8/templates/hydra.yaml.j2#L29) configs need to have some value for hydra to run, so it's fine to put `http://127.0.0.1:{HYDRA_PUBLIC_PORT}/` as a fallback in case the traefik public url is not available.

Reply:
Doesn't make much of a difference, but since it's required to have it in the `hydra.yaml` then I would argue that it should be placed in the jinja template (else the produced file is not correct).

Reply:
thx, I'll give it in another pr.